boards/shields/wnc_m14a2a/Kconfig.shield

```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2019 Linaro Limited
+# SPDX-License-Identifier: Apache-2.0
+
+config SHIELD_WNC_M14A2A
+	bool "WNC-M14A2A LTE-M Modem"
```

Comment:
Why can shields be enabled/disabled in Kconfig independently from the SHIELD_AS_LIST setting?

Reply:
This is actually directly dependent. Though (if I understand your point), we need this SHIELD_FOO symbol to enable usage of Kconfig.defconfig files.

Reply:
Though, it is right that, since shield enabling depends on SHIELD_AS_LIST, there is no way we can enable it from menuconfig (we can, but it has actually no effect) contrary to what I answered to @SebastianBoe in https://github.com/zephyrproject-rtos/zephyr/pull/20934#discussion_r349540628.
So I'm gonna review this part.

Reply:
Yeah, it's a good idea to always imagine someone going into `menuconfig` and trying really to create a broken configuration when making Kconfig changes.

Reply:
ok, removed the "menu" stuff